advocacy_docs/supported-open-source/pgbackrest/08-multiple-repositories.mdx

```diff
@@ -0,0 +1,591 @@
+            timestamp start/stop: 2021-04-19 14:46:56 / 2021-04-19 14:47:10
+            wal start/stop: 00000001000000000000001B / 00000001000000000000001B
+            database size: 63MB, database backup size: 63MB
+            repo1: backup set size: 10MB, backup size: 10MB
+
+$ rm -rf /tmp/restored_data/*
+$ pgbackrest restore --stanza=demo --target="2021-04-19 14:38:55.313716+00:00" --type=time --no-delta --pg1-path=/tmp/restored_data
+P00   INFO: repo2: restore backup set 20210419-143749F
+```
+
+The recovery process may be complex and tricky depending on the target. So, the `info` command will really be helpful for you to chose the repository from which the data should be restored.
+
+#### Archive Get Command
+
+When multiple repositories are configured, WAL will be fetched from the repositories in priority order (e.g. `repo1`, `repo2`, etc.). In general it is better if faster storage has higher priority. When a repository is specified with the `--repo` option, only that repository will be searched.
+
+```bash
+root# systemctl stop edb-as-13.service
+$ mv /var/lib/edb/as13/data /var/lib/edb/as13/data.orig
+$ mkdir -m 700 /var/lib/edb/as13/data
+
+$ pgbackrest restore --stanza=demo --target="2021-04-19 14:38:55.313716+00:00" --type=time --no-delta --target-action=promote
+P00   INFO: restore command begin ...
+P00   INFO: write updated /var/lib/edb/as13/data/postgresql.auto.conf
+P00   INFO: restore global/pg_control (performed last to ensure aborted restores cannot be started)
+P00   INFO: restore command end: completed successfully
+
+$ cat /var/lib/edb/as13/data/postgresql.auto.conf|grep restore_command
+restore_command = 'pgbackrest --stanza=demo archive-get %f "%p"'
+
+root# systemctl start edb-as-13.service
+$ cat /var/lib/edb/as13/data/log/edb-2021-04-19_145147.log
+...
+LOG:  starting point-in-time recovery to 2021-04-19 14:38:55.313716+00
+P00   INFO: found 000000010000000000000018 in the repo2: 13-1 archive
+P00   INFO: found 000000010000000000000019 in the repo2: 13-1 archive
+LOG:  recovery stopping before commit of transaction 1183, time 2021-04-19 14:38:55.314118+00
+LOG:  redo done at 0/19000100
+P00   INFO: unable to find 00000002.history in the archive
+LOG:  selected new timeline ID: 2
+LOG:  archive recovery complete
+P00   INFO: pushed WAL file '00000002.history' to the archive
+
+$ ls /var/lib/edb/as13/backups/repo*/archive/demo/13-1/00000002.history 
+/var/lib/edb/as13/backups/repo1/archive/demo/13-1/00000002.history
+/var/lib/edb/as13/backups/repo2/archive/demo/13-1/00000002.history
+```
+
+PostgreSQL found the WALs needed for recovery in our **repo2**, picked a new timeline and pushed the history file to the repositories.
+
+Every time a new timeline is created, PostgreSQL creates a timeline history file that shows which timeline it branched off from and when. These history files are necessary to allow the system to pick the right WAL segment files when recovering from a backup that contains multiple timelines.
```

Comment:
'Whenever' seems better fitting than 'Every time'.

Reply:
Fixed in 7fa8ff1